docs/http_api.rst

```diff
@@ -0,0 +1,1035 @@
+
+    Creates a new organization.
+
+    :<json str name: The name of the organization.
+    :>json str name: The name of the created organization.
+    :>json int id: The id of the created organization.
+    :>json list teams: The list of teams that the organization has.
+    :>json list users: The list of users that are part of the organization.
+    :status 201: When the organization is successfully generated.
+    :status 400: When there is invalid information to create the organization.
+
+    **Example request**:
+
+    .. sourcecode:: http
+
+       POST /organizations/ HTTP/1.1
+       Content-Type: application/json
+
+       {"name":"Nights Watch"}
+ 
+
+    **Example response**:
+
+    .. sourcecode:: http
+
+        HTTP/1.1 201 Created
+        Content-Type: application/json
+
+        {"name":"Nights Watch","id":4,"teams":[],"url":"https://example.org/organizations/4","users":[]}
+
+.. http:get:: /organizations/
+
+    Get a list of existing organizations
+
+    **Example request**:
+
+    .. sourcecode:: http
+
+       GET /organizations/ HTTP/1.1
+
+    **Example response**:
+
+    .. sourcecode:: http
+
+       HTTP/1.1 200 OK
+       Content-Type: application/json
+
+       [{"name":"Nights Watch","id":4,"teams":[],"url":"https://example.org/organizations/4","users":[]}]
+
+.. http:get:: /organizations/(int:organization_id)
+
+    Get the details of an organization.
+
+    :>json str name: The name of the created organization.
+    :>json int id: The id of the created organization.
+    :>json list teams: The list of teams that the organization has.
+
+    **Example request**:
+
+    .. sourcecode:: http
+       
+       GET /organizations/4 HTTP/1.1
+
+    **Example response**:
+
+    .. sourcecode:: http
+
+       HTTP/1.1 200 OK
+       Content-Type: application/json
+
+       {"name":"Night's Watch","id":4,"teams":[],"url":"https://example.org/organizations/4","users":[]}
+
+.. http:put:: /organizations/(int:organization_id)
+
+    Update an existing organization.
+
+    :<json str name: The name of the organization.
+    :>json str name: The name of the created organization.
+    :>json int id: The id of the created organization.
+    :>json list teams: The list of teams that the organization has.
+    :>json list users: The list of users that are part of the organization.
+    :status 200: When the organization is successfully generated.
+    :status 400: When there is invalid information to update the organization.
+
+    **Example request**:
+
+    .. sourcecode:: http
+ 
+       PUT /organizations/4 HTTP/1.1
+       Content-Type: application/json
+ 
+       {"name": "Brotherhood Without Banners"}
+ 
+    **Example response**:
+ 
+    .. sourcecode:: http
+
+       HTTP/1.1 200 OK
+       Content-Type: application/json
+
+       {"name":"Brotherhood Without Banners","id":4,"teams":[],"url":"https://example.org/organizations/4","users":[]}
+
+.. http:delete:: /organizations/(int:organization_id)
```

Comment:
Wondering if deleting organisations is something we want to support. Might result in lots of orphan data, or lots of related data being removed in the process. Maybe for now we should just not support it, and later consider supporting archiving (so they still exist, but have an `is_archived` flag we can filter on). Thoughts?